### ISSUE TITLE
Please fix the build instructions. Ubuntu 20.10 is EOL

### ISSUE THREAD

**Issue by Giszmo** [ACINQ/phoenix#240]
The build instructions fail with

```
E: The repository 'http://security.ubuntu.com/ubuntu groovy-security Release' does not have a Release file.
E: The repository 'http://archive.ubuntu.com/ubuntu groovy Release' does not have a Release file.
E: The repository 'http://archive.ubuntu.com/ubuntu groovy-updates Release' does not have a Release file.
E: The repository 'http://archive.ubuntu.com/ubuntu groovy-backports Release' does not have a Release file.
```

Ubuntu groovy is 20.10 and not supported anymore.

**Comment by emanuelb** [comment]
Using of EOL and old versions in Dockerfile including ubuntu base image version was already reported in: #165 , They respond that working on RB build environment will happen for newer version based on new codebase which currently still has only iOS support: https://github.com/ACINQ/phoenix-kmm

meanwhile I tried to compile the latest android version 1.4.20 from this repo by using newer base images:
- ubuntu via `ubuntu:rolling` image
- openjdk11 base was upgraded to latest version as well.

commands to compile:
```shell
git clone https://github.com/ACINQ/phoenix;
cd phoenix;
git checkout android-legacy-v1.4.20;
sed -i 's/ubuntu:20.10/docker.io\/ubuntu:rolling/g' Dockerfile; 
sed -i 's/adoptopenjdk\/openjdk11:jdk-11.0.12_7-alpine/docker.io\/adoptopenjdk\/openjdk11:jdk-11.0.14.1_1-alpine/g' Dockerfile; 
podman build --pull --rm -t phoenix_1420_build_apk -f Dockerfile;
podman run --rm -ti -w /home/ubuntu/phoenix phoenix_1420_build_apk
```

in container run commands:

```shell
apt-get update -y && apt-get install -y libncurses5 libcurlpp-dev;
git clone https://github.com/ACINQ/lightning-kmp;
cd lightning-kmp;
git checkout v1.2.0;
./gradlew :build;
./gradlew :publishToMavenLocal;
cd /home/ubuntu/phoenix;
sed -i '/id("kotlinx-serialization")/a id("kotlin-android-extensions")' phoenix-shared/build.gradle.kts;
./gradlew assemble;
```
which still fail with errors like (probably more changes are required to `phoenix-shared/build.gradle.kts` file):
```
e: /home/ubuntu/phoenix/phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/data/ExchangeRates.kt: (100, 25): Serializer has not been found for type '[ERROR : Instant]'. To use context serializer as fallback, explicitly annotate type or property with @Contextual
e: /home/ubuntu/phoenix/phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/utils/LogMemory.kt: (6, 8): Unresolved reference: kotlinx
```

**Comment by dpad85** [comment]
https://github.com/ACINQ/phoenix/commit/0ff688fc014cc94806daca93799821777c456e7f updated the dockerfile for the release, including the ubuntu version. You can test that against v1.4.21 once it's public.

>```./gradlew assemble;```
> which still fail with errors like (probably more changes are required to phoenix-shared/build.gradle.kts file):

Please use `./gradlew :phoenix-legacy:assemble` instead if you want to build the legacy app. The shared module is for the new, unreleased app or the iOS app. See the last command from [the legacy build instructions](https://github.com/ACINQ/phoenix/blob/master/phoenix-legacy/BUILD.md#build-the-apk).

**Comment by emanuelb** [comment]
This issue can be closed as app version 1.4.21 can be compiled with the updated Dockerfile in repo with commands:
```shell
git clone https://github.com/ACINQ/phoenix;
cd phoenix;
git checkout android-legacy-v1.4.21;
sed -i 's/ubuntu:21.10/docker.io\/ubuntu:21.10/g' Dockerfile; 
sed -i 's/eclipse-temurin:11.0.14.1_1-jdk-alpine/docker.io\/eclipse-temurin:11.0.14.1_1-jdk-alpine/g' Dockerfile; 
podman build --pull --rm -t phoenix_1421_build_apk -f Dockerfile;
podman run --rm -ti -v $(pwd):/home/ubuntu/phoenix/phoenix-legacy/build/outputs:Z -w /home/ubuntu/phoenix phoenix_1421_build_apk ./gradlew :phoenix-legacy:assemble
```
